api/gateway/share.go

```diff
@@ -111,14 +111,17 @@ func (h *Handler) getShares(ctx context.Context, height uint64, nID namespace.ID
 	return shares.Flatten(), header.Height(), err
 }
 
-func dataFromShares(shares []share.Share) ([][]byte, error) {
-	blobs, err := appshares.ParseBlobs(shares)
+func dataFromShares(input []share.Share) (data [][]byte, err error) {
+	sequences, err := shares.ParseShares(input)
```

Comment:
I would really love to dive deeper into what this does, and we might want actually to return it from our API(integrate into [Getter](https://github.com/celestiaorg/celestia-node/blob/main/share/getter.go#L29) somehow) or substitute current `Flatten` method. There might be a case where unparsed shares are not useful at all, and we might always want to return them parsed. WDYT? 

At this point, I don't have enough knowledge to understand how to move forward with this, as we want to merge this ASAP. If you don't have any ideas about this, we can make an issue and move forward with this and clean this up later.

Reply:
I think celestia-node API should continue to offer an endpoint that returns unparsed shares so that power users can interpret the raw share data however they choose to.

I think typical users will use endpoints that parse the share data back into share sequences and then the blob data that they published to a namespace.

[ParseShares](https://github.com/celestiaorg/celestia-app/blob/222bb1e0a6013efe44e910d7328c6d141fa99d1a/pkg/shares/share_merging.go#L103) uses the sequence start indicator that is written to every share to determine which chunks of shares are part of the same share sequence.

The [RawData](https://github.com/celestiaorg/celestia-app/blob/222bb1e0a6013efe44e910d7328c6d141fa99d1a/pkg/shares/share_merging.go#L145-L147) method on a share sequence extracts the actual raw data. 

Happy to dive deeper over a call. I don't think a new API endpoint is needed at the moment but will create a follow-up issue for us to discuss.

Reply:
Created https://github.com/celestiaorg/celestia-node/issues/1670

Reply:
Discussed synchronously and in this context it's appropriate to think of a share sequence as a blob. Since a namespace can contain multiple blobs, it contains multiple share sequences.